test/generic_set_access_test.py

```diff
@@ -123,51 +128,60 @@ def test_list_sets(self):
             assert "ref_path" in item
             assert item["ref_path"] == s["ref"] + ";" + item["ref"]
 
-    self.unit_test_get_set_items()
+    unit_test_get_set_items(set_data, config, set_api_client, ctx)
 
-def test_bulk_list_sets(self):
+
+def test_bulk_list_sets(
+    config: dict[str, str], clients: dict[str, Any], set_api_client: SetAPI, ctx: dict[str, str | list]
+) -> None:
+    magic_number = 1000
     try:
+        debug_lines = []
         ids = []
-        for ws_info in self.ws_client.list_workspace_info(
+        for ws_info in clients["ws"].list_workspace_info(
             {"perm": "r", "excludeGlobal": 1}
         ):
-            if ws_info[4] < 1000:
+            if ws_info[4] < magic_number:
                 ids.append(str(ws_info[0]))
             else:
-                print(f"Workspace: {ws_info[1]}, size={ws_info[4]} skipped")
+                debug_lines.append(f"Workspace: {ws_info[1]}, size={ws_info[4]} skipped")
 
-        print("Number of workspaces for bulk list_sets: " + str(len(ids)))
+        debug_lines.append("Number of workspaces for bulk list_sets: " + str(len(ids)))
         if len(ids) > 0:
-            ret = self.set_api_client.list_sets(
-                self.ctx,
+            ret = set_api_client.list_sets(
+                ctx,
                 {"workspaces": [ids[0]], "include_set_item_info": 1},
             )[0]
         GenericSetNavigator.DEBUG = True
         t1 = time.time()
-        ret = self.set_api_client.list_sets(
-            self.ctx, {"workspaces": ids, "include_set_item_info": 1}
+        ret = set_api_client.list_sets(
+            ctx, {"workspaces": ids, "include_set_item_info": 1}
         )[0]
 
-        print(f"Objects found: {len(ret['sets'])}, time={time.time() - t1}")
+        debug_lines.append(f"Objects found: {len(ret['sets'])}, time={time.time() - t1}")
+        if DEBUG:
+            log_this(config, "test_bulk_list_sets", debug_lines)
+
     finally:
         GenericSetNavigator.DEBUG = False
```

Comment:
Maybe put the `log_this` stuff here, in case anything crashes along the way? I see that's not how the original was, but this might be helpful.